lib/android/app/src/main/java/com/reactnativenavigation/views/TopBar.java

```diff
@@ -72,78 +81,136 @@ public void applyTopTabsFontSize(Number fontSize) {
-		leftBarButton.applyNavigationIcon(getContext());
-	}
-
-	private void setRightButtons(ArrayList<Button> rightButtons) {
-		if(rightButtons == null || rightButtons.size() == 0) {
-			return;
-		}
-
-		Menu menu = getTitleBar().getMenu();
-		menu.clear();
-
-		for (int i = 0; i < rightButtons.size(); i++){
-	   		Button button = rightButtons.get(i);
-			TitleBarButton titleBarButton = new TitleBarButton(container, this.titleBar, button);
-			titleBarButton.addToMenu(getContext(), menu);
-       }
-    }
-
-	public Toolbar getTitleBar() {
-		return titleBar;
-	}
+    public void setButtons(ArrayList<Button> leftButtons, ArrayList<Button> rightButtons) {
+        setLeftButtons(leftButtons);
+        setRightButtons(rightButtons);
+    }
+
+    public TextView getTitleTextView() {
+        return findTextView(titleBar);
+    }
+
+    @Override
+    public void setBackgroundColor(@ColorInt int color) {
+        titleBar.setBackgroundColor(color);
+    }
+
+    @Nullable
+    private TextView findTextView(ViewGroup root) {
+        for (int i = 0; i < root.getChildCount(); i++) {
+            View view = root.getChildAt(i);
+            if (view instanceof TextView) {
+                return (TextView) view;
+            }
+            if (view instanceof ViewGroup) {
+                return findTextView((ViewGroup) view);
+            }
+        }
+        return null;
+    }
+
+    private void setLeftButtons(ArrayList<Button> leftButtons) {
+        if (leftButtons == null || leftButtons.isEmpty()) {
+            titleBar.setNavigationIcon(null);
+            return;
+        }
+
+        if (leftButtons.size() > 1) {
+            Log.w("RNN", "Use a custom TopBar to have more than one left button");
+        }
+
+        Button leftButton = leftButtons.get(0);
+        setLeftButton(leftButton);
+    }
+
+    private void setLeftButton(final Button button) {
+        TitleBarButton leftBarButton = new TitleBarButton(container, this.titleBar, button);
+        leftBarButton.applyNavigationIcon(getContext());
+    }
+
+    private void setRightButtons(ArrayList<Button> rightButtons) {
+        if (rightButtons == null || rightButtons.size() == 0) {
+            return;
+        }
+
+        Menu menu = getTitleBar().getMenu();
+        menu.clear();
+
+        for (int i = 0; i < rightButtons.size(); i++) {
+            Button button = rightButtons.get(i);
+            TitleBarButton titleBarButton = new TitleBarButton(container, this.titleBar, button);
+            titleBarButton.addToMenu(getContext(), menu);
+        }
+    }
+
+    public Toolbar getTitleBar() {
+        return titleBar;
+    }
 
     public void setupTopTabsWithViewPager(TopTabsViewPager viewPager) {
         initTopTabs();
         topTabs.setupWithViewPager(viewPager);
     }
 
     private void initTopTabs() {
+        topTabs = new TopTabs(getContext());
         addView(topTabs);
     }
+
+    public void enableCollapse() {
```

Comment:
Can we pull this logic into `TopBarCollapser`, then invoke `topBarCollapser.enable()`?